[ECK]: 3.5.0 document securityRoles field in StackConfigPolicy (#7478)
elastic/docs-content: deploy-manage/deploy/cloud-on-k8s/elastic-stack-configuration-policies.md

Reviewer: we also need to expose this pathway in the defining roles docs: https://www.elastic.co/docs/deploy-manage/users-roles/cluster-or-deployment-auth/defining-roles#roles-management-file

I think we should also have an opinion as to the "best" path - what pattern do we expect users to follow here? does the file need to exist before the policy field? should the file be as complete as possible to begin with, or can it be a shell that is mostly modified through the policy field? etc.

Author: Added a note in the ECK section of defining-roles.md. On the "best path" question, there is no single preferred path, the choice follows the use case. Use StackConfigPolicy.securityRoles when the same roles need to target multiple clusters, the Kubernetes secrets approach for a single cluster. 29d8ba438

| `indexTemplates.composableIndexTemplates` | Configure [index templates](/manage-data/data-store/templates.md#index-templates) to define settings, mappings, and aliases that can be applied automatically to new indices.<br><br>[Specifics for index and component templates](#templates-specifics) | [Named resources map](#syntax-types)<br><br>[Index template API]({{es-apis}}operation/operation-indices-put-index-template) |
| `indexTemplates.componentTemplates` | Configure [component templates](/manage-data/data-store/templates.md#component-templates), reusable building-blocks to define settings, mappings, and aliases for new indices.<br><br>[Specifics for index and component templates](#templates-specifics) | [Named resources map](#syntax-types)<br><br>[Component template API]({{es-apis}}operation/operation-cluster-put-component-template) |
| `securityRoleMappings` | Configure [role mappings](/deploy-manage/users-roles/cluster-or-deployment-auth/mapping-users-groups-to-roles.md) to associate roles to users based on rules. | [Named resources map](#syntax-types)<br><br>[Role mapping API]({{es-apis}}operation/operation-security-put-role-mapping) |

> | `securityRoles` {applies_to}`eck: ga 3.5` | Define custom {{es}} roles that are merged into the `roles.yml` file mounted into {{es}} pods. Refer to [Specifics for security roles](#k8s-stack-config-policy-specifics-security-roles). | [Named resources map](#syntax-types)<br><br>[Security role API]({{es-apis}}operation/operation-security-put-role) |

Reviewer: small preference for explicit `+`
want to make this a little more brief - we are explaining in detail in another section anyway

the line break is to match the other "specifics" statements in the table

```suggestion
| `securityRoles` {applies_to}`eck: ga 3.5+` | Define custom {{es}} roles that are merged into the [mounted `roles.yml` file](/deploy-manage/users-roles/cluster-or-deployment-auth/defining-roles.md#roles-management-file).<br><br>[Specifics for security roles](#k8s-stack-config-policy-specifics-security-roles) | [Named resources map](#syntax-types)<br><br>[Security role API]({{es-apis}}operation/operation-security-put-role) |
```

Author: Done. d3a5d225f

          # ...
```

### Specifics for security roles [k8s-stack-config-policy-specifics-security-roles]
```{applies_to}
deployment:
  eck: ga 3.5
```

> Unlike most other policy fields, `securityRoles` is not applied through an {{es}} API. ECK merges the definitions into [the `roles.yml` file](/deploy-manage/users-roles/cluster-or-deployment-auth/defining-roles.md#roles-management-file) mounted on each {{es}} pod. ECK creates and manages this file, so no pre-existing `roles.yml` is required. {{es}} reloads that file at runtime, so changes take effect without a pod restart.

Reviewer: I'm adding the "use this when" info here too, but I'm not sure I totally understand why this approach is better for defining across clusters ... what makes it more ergonomic?

how does this option interact with the secrets-based approach? can they coexist? should we tell them to use one or the other, not both? any gotchas with transitioning from one method to the other? 

```suggestion
Unlike most other policy fields, `securityRoles` is not applied through an {{es}} API. ECK merges the definitions into [the `roles.yml` file](/deploy-manage/users-roles/cluster-or-deployment-auth/defining-roles.md#roles-management-file) mounted on each {{es}} pod. ECK creates and manages this file, so no pre-existing `roles.yml` is required. {{es}} reloads that file at runtime, so changes take effect without a pod restart.

The `securityRoles` field acts as an alternative to [file-based role management using secrets](/deploy-manage/users-roles/cluster-or-deployment-auth/defining-roles.md#roles-management-file). You should use the `securityRoles` field if you need to apply role definitions across multiple {{es}} clusters.
```

Author: They can coexist - no gotchas. The securityRoles field is additive: ECK merges the policy-defined roles into the same roles.yml file alongside any roles coming from Kubernetes secrets. The ergonomic advantage is operational consistency: if you're already using SCP to push ES cluster config, Kibana config, snapshot policies etc. across multiple clusters from one place, you can now include role definitions in that same policy resource rather than managing separate secrets per cluster. There's no recommended migration away from secrets; it's purely additive. Made the complementary relationship explicit in ad6bae3bb

> Unlike most SCP-managed settings, roles are **not** configured through the {{es}} file-based settings API (which does not support the `roles` resource). Instead, ECK merges the role definitions into the `roles.yml` file mounted into {{es}} pods. {{es}} hot-reloads this file at runtime, so role changes take effect without a pod restart.

Reviewer: we don't usually refer to the stack config policy with an acronym. perhaps try "policy" 

we can also link to the roles.yml file info and the example

```suggestion
Unlike most other policy fields, `securityRoles` is not applied through an {{es}} API. ECK merges the definitions into [the `roles.yml` file](/deploy-manage/users-roles/cluster-or-deployment-auth/defining-roles.md#roles-management-file) mounted on each {{es}} pod. {{es}} reloads that file at runtime, so changes take effect without a pod restart.

For an example, refer to [Define custom Elasticsearch roles through a policy](#k8s-stack-config-policy-security-roles-example).
```

Author: Applied 67e74d966

> Unlike most SCP-managed settings, roles are **not** configured through the {{es}} file-based settings API (which does not support the `roles` resource). Instead, ECK merges the role definitions into the `roles.yml` file mounted into {{es}} pods. {{es}} hot-reloads this file at runtime, so role changes take effect without a pod restart.

Reviewer: does this process require a roles.yml file to be present already? we should say either way.

Author: No pre-existing file needed, ECK owns the file lifecycle. Added a sentence to make that explicit. 67e74d966

          # ...
```

### Specifics for security roles [k8s-stack-config-policy-specifics-security-roles]
```{applies_to}
deployment:
  eck: ga 3.5
```

> Unlike most other policy fields, `securityRoles` is not applied through an {{es}} API. ECK merges the definitions into [the `roles.yml` file](/deploy-manage/users-roles/cluster-or-deployment-auth/defining-roles.md#roles-management-file) mounted on each {{es}} pod. ECK creates and manages this file, so no pre-existing `roles.yml` is required. {{es}} reloads that file at runtime, so changes take effect without a pod restart.

Reviewer 2: ```suggestion
ECK merges the definitions into [the `roles.yml` file](/deploy-manage/users-roles/cluster-or-deployment-auth/defining-roles.md#roles-management-file) mounted on each {{es}} pod. ECK creates and manages this file, so no pre-existing `roles.yml` is required. {{es}} reloads that file at runtime, so changes take effect without a pod restart.
```

Most other policy elements are applied through file-based settings while following an API shape. However file-based settings is an implementation detail we don't communicate to users. Maybe side step this tar pit by removing the first sentence?

          version: 1
```

### Define custom {{es}} roles through a policy [k8s-stack-config-policy-security-roles-example]
```{applies_to}
deployment:
  eck: ga 3.5
```

> Use `securityRoles` to declaratively define {{es}} roles and apply them across multiple clusters:

Reviewer: ```suggestion
Use `securityRoles` to declaratively define {{es}} roles and apply them across multiple clusters. Roles defined here are merged into `roles.yml` and hot-reloaded by {{es}} without requiring a pod restart.
```

Author: Applied. 67e74d966

```yaml
apiVersion: stackconfigpolicy.k8s.elastic.co/v1alpha1
kind: StackConfigPolicy
metadata:
  name: custom-roles-policy
spec:
  resourceSelector:
    matchLabels:
      env: production
  elasticsearch:
    securityRoles:
      my-app-reader:
        cluster:
          - monitor
        indices:
          - names:
              - logs-*
            privileges:
              - read
              - view_index_metadata
      my-app-writer:
        indices:
          - names:
              - logs-*
            privileges:
              - write
              - create_index
```

> Roles defined here are merged into `roles.yml` and hot-reloaded by {{es}} without requiring a pod restart.

Reviewer: this sentence will just get lost because it's belong a long code snippet

```suggestion
```

Author: Removed 67e74d966
